ui/components/app/import-token/token-search/token-search.component.js

```diff
@@ -1,87 +1,87 @@
-import React, { Component } from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
 import PropTypes from 'prop-types';
 import Fuse from 'fuse.js';
 import { isEqualCaseInsensitive } from '../../../../../shared/modules/string-utils';
 import { TextFieldSearch } from '../../../component-library/text-field-search/deprecated';
 import { BlockSize, Size } from '../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
 
-export default class TokenSearch extends Component {
-  static contextTypes = {
-    t: PropTypes.func,
-  };
+const getTokens = (tokenList) => Object.values(tokenList);
 
-  static defaultProps = {
-    error: null,
-    searchClassName: undefined,
-  };
+const createTokenSearchFuse = (tokenList) => {
+  return new Fuse(getTokens(tokenList), {
+    shouldSort: true,
+    threshold: 0.45,
+    location: 0,
+    distance: 100,
+    maxPatternLength: 32,
+    minMatchCharLength: 1,
+    keys: [
+      { name: 'name', weight: 0.5 },
+      { name: 'symbol', weight: 0.5 },
+    ],
+  });
+};
+export default function TokenSearch({
+  onSearch,
+  error,
+  tokenList,
+  searchClassName,
+}) {
+  const t = useI18nContext();
 
-  static propTypes = {
-    onSearch: PropTypes.func,
-    error: PropTypes.string,
-    tokenList: PropTypes.object,
-    searchClassName: PropTypes.string,
-  };
+  const [searchQuery, setSearchQuery] = useState('');
 
-  state = {
-    searchQuery: '',
-  };
+  const [tokenSearchFuse, setTokenSearchFuse] = useState(
+    createTokenSearchFuse(tokenList),
+  );
 
-  constructor(props) {
-    super(props);
-    const { tokenList } = this.props;
-    this.tokenList = Object.values(tokenList);
-    this.tokenSearchFuse = new Fuse(this.tokenList, {
-      shouldSort: true,
-      threshold: 0.45,
-      location: 0,
-      distance: 100,
-      maxPatternLength: 32,
-      minMatchCharLength: 1,
-      keys: [
-        { name: 'name', weight: 0.5 },
-        { name: 'symbol', weight: 0.5 },
-      ],
-    });
-  }
+  useEffect(() => {
+    setTokenSearchFuse(createTokenSearchFuse(tokenList));
+  }, [tokenList, searchQuery]);
```

Comment:
Hmm, why is `searchQuery` in this list of dependencies? I don't think we need to reconstruct this when the query changes

Reply:
Ah my bad, was debugging the component. removed [here](https://github.com/MetaMask/metamask-extension/pull/22548/commits/978bf0b881bd349eee7107c81cecc0068a636070)